Mathlib/Data/List/Destutter.lean

```diff
@@ -173,4 +184,185 @@ theorem destutter_eq_nil : ∀ {l : List α}, destutter R l = [] ↔ l = []
   | _ :: l => ⟨fun h => absurd h <| l.destutter'_ne_nil R, fun h => nomatch h⟩
 #align list.destutter_eq_nil List.destutter_eq_nil
 
+/--Destutter of a map is the same as the map of the destutter, as long as the map preserves the
+   relation. -/
```

Comment:
```suggestion
/-- For a relation-preserving map, `destutter'` commutes with `map`. -/
```

@YaelDillies, should there be a version for `RelHom`?

Reply:
Probably not. We should just state this for injective maps because I doubt we care about any other relation than `≠`.